gammapy/spectrum/energy_group.py

```diff
@@ -218,15 +218,16 @@ def energy_bounds(self):
 
     def find_list_idx(self, energy):
         """Find the list index corresponding to a given energy."""
+        global idx
```

Comment:
Can you remove this `global idx`?
I don't think it has any effect, and we should never use `global`, since it leads to very hard to understand code.